### DISCUSSION TITLE
how to calculate camera distance to reference plane in caustics example

### DISCUSSION THREAD

**Discussion by crz06-CK** [mitsuba-renderer/mitsuba3#944] [Q&A]
In the example : https://mitsuba.readthedocs.io/en/stable/src/inverse_rendering/caustics_optimization.html the receiving plane is postioned at a certain distance from the lens origin=[0, -7, 0] and so is the camera  origin=[0, -4.65, 0],. I just cannot find the logic how this  camera orgin is calculated. (trial n error or what's the math behind it?). The FOV of 45 doesn't cover the full receiving plane. If i change the  lens origin to say =[0, -8, 0] and keep the camera as is, the final heightmap is zoomed in. So basically my question is how to calculate the correct camera origin? 

`# Receiving plane
    'receiving-plane': {
        'type': 'obj',
        'id': 'receiving-plane',
        'filename': 'meshes/rectangle.obj',
        'to_world': \
            mi.ScalarTransform4f.look_at(
                target=[0, 1, 0],
                origin=[0, -7, 0],
                up=[0, 0, 1]
            ).scale((5, 5, 5)),
        'bsdf': {'type': 'ref', 'id': 'white-bsdf'},
    },`

`# Looking at the receiving plane, not looking through the lens
sensor_to_world = mi.ScalarTransform4f.look_at(
    target=[0, -20, 0],
    origin=[0, -4.65, 0],
    up=[0, 0, 1]
)`
![Screenshot 2023-10-17 110954](https://github.com/mitsuba-renderer/mitsuba3/assets/115268291/e53db7b5-4922-405a-be92-fe14182ae37d)
a 3d setup of the camera (yellow) at origin=[0, -4.65, 0], and fov of 45. And receiving plane (grey) at origin=[0, -7, 0]. The lens (cyan) at origin=[0, 0, 0] 
![out_plot_origin_-07_render_res_128px](https://github.com/mitsuba-renderer/mitsuba3/assets/115268291/d367bad7-7d4a-4749-92fd-81d3f4a36ac8)
lens origin=[0, -8, 0] and so is the camera  origin=[0, -4.65, 0]

![out_plot_origin_-08_render_res_128px](https://github.com/mitsuba-renderer/mitsuba3/assets/115268291/98ca37d1-ad13-448b-81de-bec12fd54e4d)
lens origin=[0, -7, 0] and so is the camera  origin=[0, -4.65, 0]

PS: any thoughts why the kernel crashes when
`render_resolution = (512, 512) `
by with low res works fine, like 
`render_resolution = (256, 256) `

**Comment by merlinND** [comment]
Hello @crz06-CK,

I don't believe that there was any special reasoning behind the position of the camera and the size of the receiving plane, the only constraint was that the field of view was fully covered by the receiving plane. Apart from that, it's pretty much arbitrary.

For the kernel crash, you will need to post the full error and stack trace. One possibility is that your GPU runs out of memory with the higher render resolution, in which case you could e.g. lower the `spp` to compensate for the increased resolution.

**Reply by crz06-CK** [reply]
Hi @merlinND , thanks for the response. I assumed that like you suggest that the only constraint is that the field of view is fully covered by the receiving plane. However when I draw it in 3d, with the set 45 degree FOV and the given distances, the projected area is well within the receiving plane (see my first image). So i fail to understand why? and how more importantly how to find a method accurately find the distance values for the camera sensor, when I move the receiving plane. Doing this manual via trial n error seems like a pain

the log when it crashes with ssp32 and 515x512px render resolution. Doesn't crash by 16 spp. What do you recon is the best setting for this application: 512x512 with 16spp or 225x256 with 32spp?

`13:32:44.423 [info] End cell 39 execution @ 1697628764422, started @ 1697628764416, elapsed time = 0.006s
13:32:44.424 [info] Kernel acknowledged execution of cell 41 @ 1697628764424
13:33:52.640 [warn] StdErr from Kernel Process jit_flush_malloc_cache(): Dr.Jit exhausted the available memory and had to flush its allocation cache to free up additional memory. This is an expensive operation and will have a negative effect on performance. You may want to change your computation so that it uses less memory. This warning will only be displayed once.
13:35:07.260 [error] Disposing session as kernel process died ExitCode: 3221226505, Reason: This version of python seems to be incorrectly compiled
(internal generated filenames are not absolute).
This may make the debugger miss breakpoints.
Related bug: http://bugs.python.org/issue1666807
jit_flush_malloc_cache(): Dr.Jit exhausted the available memory and had to flush its allocation cache to free up additional memory. This is an expensive operation and will have a negative effect on performance. You may want to change your computation so that it uses less memory. This warning will only be displayed once.

13:35:07.260 [info] Dispose Kernel process 5620.
13:35:07.283 [info] End cell 41 execution @ undefined, started @ 1697628764424, elapsed time = -1697628764.424s`
